packages/snjs/lib/services/sync/account/response_resolver.ts

```diff
@@ -6,6 +7,30 @@ import { ImmutablePayloadCollection } from '@Protocol/collection/payload_collect
 import { ImmutablePayloadCollectionSet } from '@Protocol/collection/collection_set';
 import { CopyPayload } from '@Payloads/generator';
 import { HistoryMap } from '@Lib/services/history/history_map';
+import { PayloadFormat } from '@Lib/protocol/payloads';
+
+/**
+ * Non-encrypted types are items whose values a server must be able to read.
+ * These include server extensions (such as a note history endpoint), and
+ * multi-factor authentication items, which include a secret value that the server
+ * needs to be able to read in order to enforce.
+ */
+export const NonEncryptedTypes = Object.freeze([
+  ContentType.Mfa,
+  ContentType.ServerExtension,
+]);
+
+function filterDisallowedPayloads(payloads: PurePayload[]): PurePayload[] {
+  return payloads.filter((payload) => {
+    const isEncrypted = ![
+      PayloadFormat.DecryptedBareObject,
+      PayloadFormat.DecryptedBase64String,
+    ].includes(payload.format);
```

Comment:
This has always been a weird brain twister for me 😅 I always end up translating it to _payload.format is neither DecryptedBareObject nor DecryptedBase64String_. But the code says it in a completely different order. I'm curious, how do you reason about code that you write in this way?

Reply:
Before applying the negation this can be read backwards as "format is one of x, y, z" -- with negation this becomes "format is not one of x, y, z".

What may be problematic is the negation that is applied last -- always gotta keep an eye on those.